opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFormStage.tsx

```diff
@@ -0,0 +1,206 @@
+import { PreloadedQuery, usePreloadedQuery } from 'react-relay';
+import { CircularProgress } from '@mui/material';
+import { ConnectionHandler, RecordSourceSelectorProxy } from 'relay-runtime';
+import { FormikConfig } from 'formik';
+import {
+  StixCoreRelationshipCreationFromEntityForm,
+  stixCoreRelationshipCreationFromEntityFromMutation,
+  stixCoreRelationshipCreationFromEntityQuery,
+  stixCoreRelationshipCreationFromEntityToMutation,
+  TargetEntity,
+} from './StixCoreRelationshipCreationFromEntity';
+import { StixCoreRelationshipCreationFromEntityQuery } from './__generated__/StixCoreRelationshipCreationFromEntityQuery.graphql';
+import { UseLocalStorageHelpers } from '../../../../utils/hooks/useLocalStorage';
+import { commitMutation, handleErrorInForm } from '../../../../relay/environment';
+import { isNodeInConnection } from '../../../../utils/store';
+import { formatDate } from '../../../../utils/Time';
+import { UserContext } from '../../../../utils/hooks/useAuth';
+import { resolveRelationsTypes } from '../../../../utils/Relation';
+import StixCoreRelationshipCreationForm from './StixCoreRelationshipCreationForm';
+import { CreateRelationshipContext } from './CreateRelationshipContextProvider';
+
+interface StixCoreRelationshipCreationFormStageProps {
+  targetEntities: TargetEntity[];
+  queryRef: PreloadedQuery<StixCoreRelationshipCreationFromEntityQuery, Record<string, unknown>>;
+  handleReverseRelation?: () => void;
+  handleResetSelection: () => void;
+  handleClose: () => void;
+  defaultStartTime: string;
+  defaultStopTime: string;
+  helpers: UseLocalStorageHelpers;
+  entityId: string;
+}
+
+const StixCoreRelationshipCreationFormStage: FunctionComponent<StixCoreRelationshipCreationFormStageProps> = ({
+  targetEntities,
+  queryRef,
+  handleReverseRelation,
+  handleResetSelection,
+  handleClose,
+  defaultStartTime,
+  defaultStopTime,
+  helpers,
+  entityId,
+}) => {
+  const { stixCoreObject } = usePreloadedQuery(
+    stixCoreRelationshipCreationFromEntityQuery,
+    queryRef,
+  );
+
+  const { state: {
+    relationshipTypes: allowedRelationshipTypes,
+    reversed: isRelationReversed,
+    paginationOptions,
+    connectionKey,
+    onCreate,
+  } } = useContext(CreateRelationshipContext);
+
+  if (!stixCoreObject) {
+    return (
+      <div style={{ display: 'table', height: '100%', width: '100%' }}>
+        <span
+          style={{
+            display: 'table-cell',
+            verticalAlign: 'middle',
+            textAlign: 'center',
+          }}
+        >
+          <CircularProgress size={80} thickness={2} />
+        </span>
+      </div>
+    );
+  }
+
+  const sourceEntity: TargetEntity = stixCoreObject;
+  let fromEntities = [sourceEntity];
+  let toEntities = targetEntities;
+  if (isRelationReversed) {
+    fromEntities = targetEntities;
+    toEntities = [sourceEntity];
+  }
+
+  const commit = (finalValues: object) => {
+    return new Promise((resolve, reject) => {
```

Comment:
don't need to create a promise here, you can directly call commitMutation in the try of onSubmit